src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -38,59 +38,40 @@ const propTypes = {
         /** The authToken for the current session */
         email: PropTypes.string,
     }),
-
-    /** Beta features list */
-    betas: PropTypes.arrayOf(PropTypes.string),
 };
 
 const defaultProps = {
     route: {
         params: {},
     },
     session: {},
-    betas: null,
 };
 
 class LogInWithShortLivedTokenPage extends Component {
     componentDidMount() {
-        const accountID = parseInt(lodashGet(this.props.route.params, 'accountID', ''), 10);
-        const email = lodashGet(this.props.route.params, 'email', '');
-        const shortLivedToken = lodashGet(this.props.route.params, 'shortLivedToken', '');
-
-        // User is trying to transition with a different account than the one they are currently signed in as so we will sign out and then sign in with the new authToken
-        if (email !== this.props.session.email) {
-            Session.signOut();
-            Session.signInWithShortLivedToken(accountID, email, shortLivedToken);
+        if (this.signInIfNeeded()) {
             return;
         }
 
-        // If the user is revisiting the component authenticated with the right account, we don't need to do anything, the componentWillUpdate when betas are loaded and redirect
-        if (this.props.session.authToken && email === this.props.session.email) {
-            return;
-        }
-
-        Session.signInWithShortLivedToken(accountID, email, shortLivedToken);
+        this.signOutIfNeeded();
     }
 
     componentDidUpdate() {
-        if (!this.props.betas || !this.props.session.authToken) {
+        if (!this.props.session.authToken) {
             return;
         }
 
-        const email = lodashGet(this.props.route.params, 'email', '');
-
-        // exitTo is URI encoded because it could contain a variable number of slashes (i.e. "workspace/new" vs "workspace/<ID>/card")
-        const exitTo = decodeURIComponent(lodashGet(this.props.route.params, 'exitTo', ''));
+        if (this.signInIfNeeded()) {
```

Comment:
I think this cleanup would maybe would be better in a separate follow up PR. 
Was pretty close to merging this code, but honestly not too sure about the latest changes...

Reply:
1. It makes this change harder to track and understand in isolation. 
2. I find it strange that `componentDidMount()` was signing out first then signing in. And `componentDidUpdate()` never needed a check to sign in and we are now adding one... so I question the necessity of that.

Basically, I have to think about these changes more (I don't really like thinking more than I have to) and would rather think about whether this component needs to be improved further or not (I'm not sure it does).

Reply:
> 1. It makes this change harder to track and understand in isolation.

I agree. The same is true for adding a signOutIfNeeded function. I could do both of these in a follow up PR.

> 2. I find it strange that `componentDidMount()` was signing out first then signing in (now looks to be doing the opposite). And `componentDidUpdate()` never needed a check to "sign in" and we are now adding one... so I question the necessity of that.

componentDidMount() has been signing in first and then signing out, but really it only does on or the other. [This](https://github.com/Expensify/App/blob/4fbbdd83e0e93eca1b749210b8d8f036ba596b25/src/pages/LogInWithShortLivedTokenPage.js#L60-L64) is what I had before the refactor.

componentDidUpdate() always had a check to sign in, both [before all my changes](https://github.com/Expensify/App/blob/4cbd7ff86cfed6214a01231208fa2555269b403b/src/pages/LogInWithShortLivedTokenPage.js#L101-L107) and [before my refactor](https://github.com/Expensify/App/blob/4fbbdd83e0e93eca1b749210b8d8f036ba596b25/src/pages/LogInWithShortLivedTokenPage.js#L86). That being said I'm not totally sure if it is necessary to check if the user is signed in here.

Please let me know what you think. If this still seems like too many changes then I'll go back to what I had in commit 39bd7d5f977d11074552a2a3a3194a14f225531f, and I'll keep the commit with the removal of the betas.